matrix_factorization/tests/test_sgd.cu

```diff
@@ -53,11 +53,38 @@ void test_sgd() {
     cudaMalloc(&errors_device, matrix->nonzeros * sizeof(float));
     cudaMemcpy(errors_device, errors, matrix->nonzeros * sizeof(float), cudaMemcpyHostToDevice);
 
+    // Create the bias arrays
+    float *user_bias = new float[rows];
```

Comment:
Since we're also training user bias, we should check if the updated biases are correct (see line 70).

Reply:
kk, ill find a way to test for that